index.js

```diff
@@ -4,7 +4,40 @@
  * Depending on the operation, either add up all of the numbers or subtract all of the numbers, from left to right.
  * @returns {number} The result of either adding all numbers or subtracting all numbers, depending on the arguments added to the command line.
  */
-function calculator() {}
+function calculator() {
+   //console.log(process.argv)
+    
+   if(!process.argv[3]) {
+        
+    if(!process.argv[2]) {
+        return "No operation provided...";
+}
+return "No numbers provided...";
+}
+if( process.argv[2] === 'plus'){
+    let sum = 0
+        for(let i = 3; i < process.argv.length; i++){
+            
+         sum = sum + Number(process.argv[i])     
+        }
+        return sum
+}
 
-// Don't change anything below this line.
+ if( process.argv[2] === 'minus'){ 
+    let sum1 = process.argv[3]
+    for(let i = 4; i < process.argv.length; i++){
+    
+     sum1  -= Number(process.argv[i])
+
+    }  
+    return sum1
+}
+
+if (process.argv[2] !== 'plus' || process.argv[2] !== 'minus' ){
```

Comment:
Before you jump into adding and subtracting your inputs (line 17 - 34), you want to make sure that all the essentials for the arithmetic are present. Therefore, you would want to do your error handling first. 

i would update the control flow here by moving this further up in my code.